common/util/util.go

```diff
@@ -143,3 +143,12 @@ func FilterSlice[T any](in []T, predicate func(T) bool) []T {
 	}
 	return out
 }
+
+// ReduceSliceInitial reduces a slice using given reducer function and initial value.
+func ReduceSliceInitial[T any, A any](in []T, initializer A, reducer func(A, T) A) A {
```

Comment:
```suggestion
func ReduceSlice[T any, A any](in []T, initializer A, reducer func(A, T) A) A {
```
Reduce always needs an initial value, no need to include it in the name

Reply:
There's also reduce from T[] to T that can use the first element as the initializer. I didn't add it to our "godash" lib though.

Reply:
I'd rather force callers to do reduce(slice[1:], slice[0], f) so then it's clearly the caller's responsibility to do something for the empty slice

Reply:
okay